dotnet/src/SemanticKernel.Abstractions/Kernel.cs

```diff
@@ -176,6 +177,12 @@ public CultureInfo Culture
         set => this._culture = value ?? CultureInfo.InvariantCulture;
     }
 
+    /// <summary>
+    /// Gets the Json serializer options currently associated with this <see cref="Kernel"/>.
+    /// </summary>
+    [Experimental("SKEXP0001")]
+    public JsonSerializerOptions? SerializerOptions { get; set; }
```

Comment:
@eiriktsarpalis, can you share insights here on what you'd expect to be passed throughout a system like this?

Reply:
JsonSerializerOptions controls every aspect of serialization, including what types can be serialized (in the context of source generators).

@SergeyMenshykh assuming that this options instance is used to serialize types that are internal to SK, this could create problems if and when you'd be looking to add Native AOT support: user-supplied source generated JSO instances cannot support SK internals.

ASP.NET Core addresses this issue by making their options properties get-only (ensuring that they are created and owned by them) and then apply fix-up logic to make sure that the instance works with their own internal models.

Reply:
@eiriktsarpalis, we'll need to enable consumers to supply their own and then presumably merge with internal information in order to serialize both user and internal types. What is the recommended way to do that? I don't understand how making this property get-only would enable that. What shoukdbe exposed here to allow a consumer to contribute type information?

Reply:
> What is the recommended way to do that?

The `JsonSerializerOptions.TypeInfoResolverChain` property is an accelerator API that lets you combine multiple source generated contexts in one instance. ASP.NET uses that API to append their own stuff after users have finished configuring the object.

> I don't understand how making this property get-only would enable that.

The property can be set to shared, read-only instances such as `JsonSerializerOptions.Default`. Arguably this can be worked around using the `JsonSerializerOptions` copy constructor and applying post processing steps directly inside the setter body, but a getter-only approach better communicates that the particular options (and its associated caches) are owned by the `Kernel` instance.

> What should be exposed here to allow a consumer to contribute converter and type information?

I don't think we need to expose anything other than the APIs already present in the `JsonSerializerOptions` itself.

Reply:
I'm saying that this should be designed in way where users can configure a JSO instance that is owned by the kernel, but which can then be followed by a post-processing phase where source generated contexts defined internally in SK can be appended to user configuration. I don't think this could be achieved effectively with a property that can be mutated at any point in the lifetime of the kernel instance. ASP.NET Core does it using their configuration builder APIs.

Of course, I could be completely wrong in my original assumption that internal SK types are being serialized using this configuration in which case it doesn't matter a lot.